### PR TITLE
PLAT-168: read Confluent Cloud configuration

### PR DESCRIPTION
**Pull request by ybyzek**
https://confluentinc.atlassian.net/browse/PLAT-168

By request, I validated this works on both macos and linux

### CODE REVIEW

File: src/oss/confluent.sh [hunk 1]
```diff
@@ -1697,21 +1716,22 @@ consume_usage() {
...
     To consume from another Kafka cluster, set the '--bootstrap-server' argument.
+    To consume Confluent Cloud, set the '--cloud' argument (assumes Confluent Cloud configuration file at $HOME/.ccloud/config and topic already created in Confluent Cloud).
 
     By default, this command reads non-Avro data.
     To read Avro data, specify '--value-format avro'
 
 Examples:
     confluent consume mytopic1 --value-format avro --from-beginning
 
-    confluent consume mytopic1 --from-beginning
```
Comment (kkonstantine) [inline src/oss/confluent.sh]: same as with produce above, we probably want to retain the original example with minimal parameters that refers to a local cluster.

Comment (kkonstantine) [reply src/oss/confluent.sh]: In both cases I'd have this example first too, to align with the order mentioned in the description.

File: src/oss/confluent.sh [hunk 2]
```diff
@@ -1497,9 +1497,17 @@ produce_command() {
       export_kafka
       bootstrapserver="--broker-list localhost:${kafka_port}"
     fi
-    removestr="--value-format avro"
-    if [[ "$args" =~ "$removestr" ]]; then
-      args=${args//$removestr/}
+    cloud="--cloud"
+    if [[ "$args" =~ "$cloud" ]]; then
+      file="$HOME/.ccloud/config"
```
Comment (kkonstantine) [inline src/oss/confluent.sh]: this is hardcoded here and it won't work if you have two ccloud clusters I believe. ccloud accepts the config as a parameter. Thoughts?

File: src/oss/confluent.sh [hunk 3]
```diff
@@ -1497,9 +1497,17 @@ produce_command() {
...
+      args=${args//$cloud/--producer.config $file}
+      BOOTSTRAP_SERVERS=$( grep "^bootstrap.server" $file | awk -F'=' '{print $2;}' )
```
Comment (kkonstantine) [inline src/oss/confluent.sh]: awk is not very portable
https://github.com/confluentinc/confluent-cli/blob/master/src/oss/confluent.sh#L1365-L1366
you may see a two line example of how to extract a value from a key-value line where the pair is separated by `=`

File: src/oss/confluent.sh [hunk 4]
```diff
@@ -1497,9 +1497,17 @@ produce_command() {
...
+      BOOTSTRAP_SERVERS=${BOOTSTRAP_SERVERS/\\/}
```
Comment (kkonstantine) [inline src/oss/confluent.sh]: If this is needed because of ccloud config format we'd need a comment here.

File: src/oss/confluent.sh [hunk 5]
```diff
@@ -1521,9 +1529,17 @@ consume_command() {
       export_kafka
       bootstrapserver="--bootstrap-server localhost:${kafka_port}"
     fi
-    removestr="--value-format avro"
-    if [[ "$args" =~ "$removestr" ]]; then
-      args=${args//$removestr/}
+    cloud="--cloud"
+    if [[ "$args" =~ "$cloud" ]]; then
```
Comment (kkonstantine) [inline src/oss/confluent.sh]: basically this is the same code as above, just with a different param. Ideally we would factor out in a function and call it in both places with producer or consumer arguments. See `extract_name_from_properties_file` and how it's used for a very similar example of such a function.

File: src/oss/confluent.sh (hunk 1 above)
Comment (ybyzek) [reply src/oss/confluent.sh]: Agree

File: src/oss/confluent.sh (hunk 2 above)
Comment (ybyzek) [reply src/oss/confluent.sh]: I like the simplicity of just specifying the `--cloud` argument.

I'm not against specifying the ccloud config file, but do you happen to know how many users have multiple ccloud clusters?

File: src/oss/confluent.sh (hunk 5 above)
Comment (ybyzek) [reply src/oss/confluent.sh]: Great suggestion!

File: src/oss/confluent.sh (hunk 4 above)
Comment (ybyzek) [reply src/oss/confluent.sh]: Agree

File: src/oss/confluent.sh (hunk 3 above)
Comment (ybyzek) [reply src/oss/confluent.sh]: Sounds good

File: src/oss/confluent.sh (hunk 2 above)
Comment (ybyzek) [reply src/oss/confluent.sh]: I added this in.  Not super robust, but it works.

**Comment by ybyzek** [comment]
@kkonstantine thank you so much for your review so far.  I have addressed your feedback, hopefully well ;)

**Pull request merged**